### PR TITLE
Fix lint errors and add typings

### LINKED ISSUE #29: Clean up ESLint + type errors
## Tasks

- [x] #34
- [ ] #47

### PR DESCRIPTION
**Pull request by jackyef**
Closes #29

### CODE REVIEW

**Comment by tibudiyanto** [comment]
This has conflicts @jackyef

**Review by resir014 — APPROVED** [review]
This looks good! Please resolve all the merge conflicts before I can merge this.

**Comment by jackyef** [comment]
The PR has grown in size since there are quite a few additions to the `main` branch. Please re-review.

Things looks good locally.
```sh
➜  jakarta-vax-availability git:(lint-ts-error) yarn lint && yarn tsc
yarn run v1.22.10
$ eslint "**/*.{js,jsx,ts,tsx}"
✨  Done in 11.33s.
yarn run v1.22.10
$ /Users/jackyef/Personal/Projects/jakarta-vax-availability/node_modules/.bin/tsc
✨  Done in 6.11s.
```

File: src/components/VaxLocation.tsx [hunk 1]
```diff
@@ -21,15 +22,29 @@ import {
...
 } from '@chakra-ui/react';
+import { DetailLokasi, VaccinationData } from 'data/types';
 import { formatDistanceToNow } from 'date-fns';
 import idLocale from 'date-fns/locale/id';
-import { ExternalLinkIcon } from "@chakra-ui/icons";
 
-export default function VaxLocation({ loading, location, isUserLocationExist }) {
+type DetailLokasiWithDistance = DetailLokasi & {
+  distance?: string | null;
+};
```
Comment (pveyes) [inline src/components/VaxLocation.tsx]: Personally I prefer interface for things like this

```suggestion
interface DetailLokasiWithDistance extends DetailLokasi {
  distance?: string | null;
};
```

File: src/components/VaxLocation.tsx [hunk 2]
```diff
@@ -21,15 +22,29 @@ import {
...
+
+export type VaccinationDataWithDistance = Omit<VaccinationData, 'detail_lokasi'> & {
+  detail_lokasi?: (DetailLokasiWithDistance | null)[] | null;
+};
```
Comment (pveyes) [inline src/components/VaxLocation.tsx]: Same as above

```suggestion
export interface VaccinationDataWithDistance extends Omit<VaccinationData, 'detail_lokasi'> {
  detail_lokasi?: (DetailLokasiWithDistance | null)[] | null;
};
```

File: src/pages/map.tsx [hunk 3]
```diff
@@ -49,7 +52,14 @@ const Map = ReactMapboxGl({
   accessToken: process.env.NEXT_PUBLIC_MAPBOX_KEY
 });
 
-const Mark = ({ lokasi }) => (
+type LocationData = Partial<DetailLokasi> & {
+  jadwal?: Jadwal[] | null;
+};
```
Comment (pveyes) [inline src/pages/map.tsx]: use interface 

```suggestion
interface LocationData extends Partial<DetailLokasi> {
  jadwal?: Jadwal[] | null;
};
```

File: src/components/VaxLocation.tsx [hunk 4]
```diff
@@ -21,15 +22,29 @@ import {
...
 import idLocale from 'date-fns/locale/id';
-import { ExternalLinkIcon } from "@chakra-ui/icons";
 
-export default function VaxLocation({ loading, location, isUserLocationExist }) {
+type DetailLokasiWithDistance = DetailLokasi & {
+  distance?: string | null;
+};
+
+export type VaccinationDataWithDistance = Omit<VaccinationData, 'detail_lokasi'> & {
+  detail_lokasi?: (DetailLokasiWithDistance | null)[] | null;
```
Comment (pveyes) [inline src/components/VaxLocation.tsx]: btw can we refine this type? it feels too much

would be better if it's just `detail_lokasi?: DetailLokasiWithDistance[]`

File: src/components/VaxLocation.tsx (hunk 1 above)
Comment (jackyef) [reply src/components/VaxLocation.tsx]: Added  in https://github.com/tibudiyanto/jakarta-vax-availability/pull/34/commits/6259a467af374b737231680a0d41c5ce906323d4

File: src/components/VaxLocation.tsx (hunk 2 above)
Comment (jackyef) [reply src/components/VaxLocation.tsx]: Added  in https://github.com/tibudiyanto/jakarta-vax-availability/pull/34/commits/6259a467af374b737231680a0d41c5ce906323d4

File: src/pages/map.tsx (hunk 3 above)
Comment (jackyef) [reply src/pages/map.tsx]: Added  in https://github.com/tibudiyanto/jakarta-vax-availability/pull/34/commits/6259a467af374b737231680a0d41c5ce906323d4

File: src/components/VaxLocation.tsx (hunk 4 above)
Comment (jackyef) [reply src/components/VaxLocation.tsx]: Added  in https://github.com/tibudiyanto/jakarta-vax-availability/pull/34/commits/6259a467af374b737231680a0d41c5ce906323d4

**Review by pveyes — APPROVED** [review]
LGTM assuming green CI cc @tibudiyanto for auth

**Review by resir014 — APPROVED** [review]
Alright, looks good. Thanks!

**Pull request merged**